mantidproject/mantid
# Indirect Edit Results Option in ConvFit and IqtFit

**Description of work.**
During a sequential fit in Indirect Data Analysis, the end parameters of one spectrums fit become the start parameters for the next spectrum. This can be a problem if the next spectrum is not related to the previous spectrum and will lead to a bad fit.
This PR adds an `Edit Result` option which allows you to select a result workspace with multiple bins (produced by fitting multiple spectra at a time), and a result workspace with a single bin (produced by fitting a single spectrum) and then replace the corresponding values with the same Q in the larger workspace with the bins found in the single-bin workspace.

The output workspace is placed back into the result workspace group containing the workspace with multiple bins.

This option is only available for ConvFit and IqtFit.

**To test:**
1.`Interfaces`->`Indirect`->`Data Analysis`->`ConvFit` tab
2. Load the data below 
3. Select Fit Type to be One Lorentzian
4. Click `Run` and wait
5. Then select the spectrum selected to be `2` and click `Fit Single Spectrum`
6. Click `Edit Result` and a dialog box should appear.
7. Make sure the two output _Result workspaces are selected
8. Choose a name for your output workspace (make it different to the input workspace)
9. Click `Replace Bin`

Check the large _Results group workspace for the outputted workspace, and open this workspace. Open the original input result workspace and the result workspace corresponding to a single fit.

Compare the values in the column labelled `2` in the output workspaces with the values of the single fit result workspace - they should be the same.
Make sure that the values in the column labelled `2` in the output workspaces are different to those in the original input result workspace.

Repeat for IqtFit. Check to make sure invalid data can't be entered. Read the documentation to make sure it makes sense.

**Data Files**
ConvFit:
[iris26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2841458/iris26176_graphite002_red.zip)
[iris26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2841459/iris26173_graphite002_res.zip)

IqtFit:
[irs26176_graphite002_iqt.zip](https://github.com/mantidproject/mantid/files/2844695/irs26176_graphite002_iqt.zip)
2 Exponentials and slide the endx slider slightly to the left

Fixes #24691 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
